definitions/npm/lodash_v4.x.x/flow_v0.104.x-/lodash_v4.x.x.js

```diff
@@ -570,61 +575,67 @@ declare module "lodash" {
       predicate?: ?Predicate<T>,
       fromIndex?: ?number
     ): void;
-    find<V, A, T: { [id: any]: A, ... }>(
+    find<V, A, T: ReadOnlyIndexerObject<A>>(
       object: T,
-      predicate?: OPredicate<A, T>,
-      fromIndex?: number
+      predicate?: ?OPredicate<A, T>,
+      fromIndex?: ?number
     ): V;
     findLast<T>(
-      array: ?$ReadOnlyArray<T>,
+      array: $ReadOnlyArray<T>,
       predicate?: ?Predicate<T>,
       fromIndex?: ?number
     ): T | void;
-    findLast<V, A, T: { [id: any]: A, ... }>(
+    findLast<T>(
+      array: void | null,
+      predicate?: ?Predicate<T>,
+      fromIndex?: ?number
+    ): void;
+    findLast<V, A, T: ReadOnlyIndexerObject<A>>(
       object: T,
-      predicate?: ?OPredicate<A, T>
+      predicate?: ?OPredicate<A, T>,
+      fromIndex?: ?number
     ): V;
     flatMap<T, U>(
       array?: ?$ReadOnlyArray<T>,
-      iteratee?: ?FlatMapIteratee<T, U>
+      iteratee?: ?AFlatMapIteratee<T, U>
     ): Array<U>;
-    flatMap<T: Object, U>(
+    flatMap<A, T: ReadOnlyIndexerObject<A>, U>(
       object: T,
-      iteratee?: OFlatMapIteratee<T, U>
+      iteratee?: OFlatMapIteratee<A, T, U>
     ): Array<U>;
     flatMapDeep<T, U>(
       array?: ?$ReadOnlyArray<T>,
-      iteratee?: ?FlatMapIteratee<T, U>
+      iteratee?: ?AFlatMapIteratee<T, U>
     ): Array<U>;
-    flatMapDeep<T: Object, U>(
+    flatMapDeep<A, T: ReadOnlyIndexerObject<A>, U>(
       object: T,
-      iteratee?: ?OFlatMapIteratee<T, U>
+      iteratee?: ?OFlatMapIteratee<A, T, U>
     ): Array<U>;
     flatMapDepth<T, U>(
       array?: ?$ReadOnlyArray<T>,
-      iteratee?: ?FlatMapIteratee<T, U>,
+      iteratee?: ?AFlatMapIteratee<T, U>,
       depth?: ?number
     ): Array<U>;
-    flatMapDepth<T: Object, U>(
+    flatMapDepth<A, T: ReadOnlyIndexerObject<A>, U>(
       object: T,
-      iteratee?: OFlatMapIteratee<T, U>,
+      iteratee?: OFlatMapIteratee<A, T, U>,
       depth?: number
     ): Array<U>;
     forEach<T>(array: $ReadOnlyArray<T>, iteratee?: ?Iteratee<T>): Array<T>;
     forEach<T: void | null>(array: T, iteratee?: ?Iteratee<any>): T;
-    forEach<T: Object>(object: T, iteratee?: ?OIteratee<T>): T;
+    forEach<A, T: ReadOnlyIndexerObject<A>>(object: T, iteratee?: ?IterateeWithResult<A, T, boolean | void>): T;
     forEachRight<T>(
       array: $ReadOnlyArray<T>,
       iteratee?: ?Iteratee<T>
     ): Array<T>;
     forEachRight<T: void | null>(array: T, iteratee?: ?Iteratee<any>): T;
-    forEachRight<T: Object>(object: T, iteratee?: ?OIteratee<T>): T;
+    forEachRight<A, T: ReadOnlyIndexerObject<A>>(object: T, iteratee?: ?IterateeWithResult<A, T, boolean | void>): T;
     groupBy<V, T>(
       array: $ReadOnlyArray<T>,
       iteratee?: ?ValueOnlyIteratee<T>
     ): { [key: V]: Array<T>, ... };
     groupBy(array: void | null, iteratee?: ?ValueOnlyIteratee<any>): {...};
-    groupBy<V, A, T: { [id: any]: A, ... }>(
+    groupBy<V, A, T: ReadOnlyIndexerObject<A>>(
```

Comment:
Btw, I'm not sure why A is used for value type, I'd use V for it. But was following current naming convention for consistency